tests/test_config_parsing/config_dict_generator.py

```diff
@@ -280,37 +287,27 @@ def generate_config(draw):
                     "gen-kw-export-name-" + draw(file_names)
                 ),
                 ConfigKeys.FIELD_KEY: st.lists(
-                    st.fixed_dictionaries(
-                        {
-                            ConfigKeys.NAME: st.just("FIELD-" + draw(words)),
-                            ConfigKeys.VAR_TYPE: st.just("PARAMETER"),
-                            ConfigKeys.OUT_FILE: file_names,
-                            # ConfigKeys.ENKF_INFILE: file_names, only used in general
-                            ConfigKeys.FORWARD_INIT: st.booleans(),
-                            ConfigKeys.INIT_TRANSFORM: transforms,
-                            ConfigKeys.OUTPUT_TRANSFORM: transforms,
-                            # ConfigKeys.INPUT_TRANSFORM: func, only used in general
-                            ConfigKeys.MIN_KEY: small_floats,
-                            ConfigKeys.MAX_KEY: small_floats,
-                            ConfigKeys.INIT_FILES: file_names,
-                        }
+                    st.tuples(
+                        st.just("FIELD-" + draw(words)),
+                        st.just("PARAMETER"),
+                        file_names,
+                        st.just(f"FORWARD_INIT:{draw(st.booleans())}"),
+                        st.just(f"INIT_TRANSFORM:{draw(transforms)}"),
+                        st.just(f"OUTPUT_TRANSFORM:{draw(transforms)}"),
+                        st.just(f"MIN:{draw(small_floats)}"),
+                        st.just(f"MAX:{draw(small_floats)}"),
+                        st.just(f"INIT_FILES:{draw(file_names)}"),
                     ),
-                    unique_by=lambda field_dict: field_dict[ConfigKeys.NAME],
+                    unique_by=lambda element: element[0],
                 ),
                 ConfigKeys.GEN_DATA: st.lists(
-                    st.fixed_dictionaries(
-                        {
-                            ConfigKeys.NAME: st.just("GEN_DATA-" + draw(words)),
-                            ConfigKeys.RESULT_FILE: format_file_names,
-                            ConfigKeys.INPUT_FORMAT: st.just(GenDataFileType.ASCII),
-                            ConfigKeys.REPORT_STEPS: st.lists(
-                                st.integers(min_value=0, max_value=100),
-                                min_size=2,
-                                unique=True,
-                            ),
-                        }
+                    st.tuples(
```

Comment:
only one question: you use tuples here - why not lists, like in the _c_wrappers tests?

Reply:
Hypothesis does not like nested lists with complex elements inside them or it is much harder to get the syntax right if you want to have them like that.

Reply:
ok cool thanks!